platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -555,16 +576,18 @@ export default class HangingProtocolService extends PubSubService {
     };
   }
 
-  getViewportsRequireUpdate(viewportIndex, displaySetInstanceUID) {
+  getViewportsRequireUpdate(viewportId, displaySetInstanceUID) {
     const newDisplaySetInstanceUID = displaySetInstanceUID;
     const protocol = this.protocol;
     const protocolStage = protocol.stages[this.stageIndex];
     const protocolViewports = protocolStage.viewports;
-    const protocolViewport = protocolViewports[viewportIndex];
+    const protocolViewport = protocolViewports.find(
```

Comment:
As a sanity check, consider checking if `protocolViewport` is `undefined` before proceeding.